### DISCUSSION TITLE
Removing default `HomePage` bricks leptos

### DISCUSSION THREAD

**Discussion by Kiwifuit** [leptos-rs/leptos#2860] [Q&A]
I tried removing the `<Route path="" view=HomePage/>` line and the whole function block for `HomePage` and I get this thread panic when I run `cargo leptos watch`
```
thread 'main' panicked at /home/misery/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_reactive-0.6.14/src/context.rs:316:9:
Location { file: "mcs/src/app.rs", line: 52, col: 20 } expected context of type "leptos_actix::ResponseOptions" to be present
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/thread/local.rs:260:26:
cannot access a Thread Local Storage value during or after destruction: AccessError
fatal runtime error: thread local panicked on drop
```
Why does it do this?

**Comment by gbj** [comment]
Reading between the lines, you have started with the Actix started and removed the only route, so it runs the 404 during route generation on the initial server setup. During route generation, no `ResponseOptions` are provided because there is no HTTP response, so the `expect_context()` in the `NotFound` page fails.

You can avoid the panic pretty straightforwardly with
```rust
        if let Some(resp) = use_context::<leptos_actix::ResponseOptions>() {
            resp.set_status(actix_web::http::StatusCode::NOT_FOUND);
        }
```
(this is defined in user code)

There's no reason you'd ever encounter this other than deleting all possible routes.
